batch-jberet/src/main/java/org/wildfly/extension/batch/jberet/deployment/JobOperatorService.java

```diff
@@ -432,7 +448,27 @@ public void suspended(final ServerActivityCallback serverActivityCallback) {
 
         @Override
         public void resume() {
-            restartStoppedJobs();
+            boolean doResume;
+            synchronized (this) {
+                suspended = false;
+                doResume = running;
+            }
+            if (doResume) {
+                restartStoppedJobs();
+            }
+        }
+
+        @Override
+        public void propertyChange(PropertyChangeEvent evt) {
+            boolean doResume;
+            synchronized (this) {
+                ControlledProcessState.State newState = (ControlledProcessState.State) evt.getNewValue();
+                running = newState.isRunning();
+                doResume = running && !suspended;
+            }
+            if (doResume) {
+                restartStoppedJobs();
+            }
         }
```

Comment:
@chengfang FYI. What this is about is the fact that with this RFE the behavior of boot can be different, where users can configure 'graceful startup' to be turned off. With graceful startup this kind of sequence happens

server is initially in a suspended state
boot is 99% done, boot logic 'resumes' the server to move out of suspended. The resume() method above is called
server switches from ControlledProcessState.State.STARTING to RUNNING (i.e. boot is done)

The gap between the last two bits is miniscule.

If the user has disable graceful startup, the server is not initially in suspended state, so no resume call goes out. So no signal that triggers 'restartStoppedJobs' will be received. So this change is adding a listener for when the ControlledProcessState.State.STARTING to RUNNING change happens and uses that to ensure  restartStoppedJobs is called.

If after boot the server is suspend and later resumed, that will result in a call to restartStoppedJobs, same as before.